### PR TITLE
Fix the need for an exact path to be used

### LINKED ISSUE #47: Absolute Path of Command is Required on MacOS when Terminal Mode is Off
### **Describe the bug:**

On MacOS systems, (as mentioned in the bottom of the README) an absolute path is required to run commands when terminal mode is off.

### **To Reproduce:**

1. Set the `command` to use a non built-in command, such as `"node ${path.join(__dirname, './tester.js')} $_URL_"` (See how it uses node?)
2. Set `terminal` to `false` in the configuration.
3. Register the protocol.
4. Go to the protocol in a browser.
5. It will error and say that the command does not exist.

### **Expected behavior:**

It should run the command like a user is doing so.

<hr/>

### PR DESCRIPTION
**Pull request by Grayseon**
# Description:

Changes any nonspecific paths to be exact.

Fixes #47 

## Type of change:

-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] This change requires a documentation update

# Checklist:

-   [x] My code follows the style guidelines of this project.
-   [x] I have performed a self-review of my own code.
-   [x] I have commented my code, particularly in hard-to-understand areas.
-   [x] I have made corresponding changes to the documentation.
-   [x] My changes generate no new warnings.
-   [x] I have added tests that prove my fix is effective or that my feature works.
-   [x] New and existing unit tests pass locally with my changes.
-   [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: src/utils/replaceWithExactPath.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const which = require('which');
+
+function replace(command) {
+    return command
+        .split('&&')
```
Comment (Shubham-Kumar-2000) [inline src/utils/replaceWithExactPath.js:5]: commands can also be split by `;`

Comment (Shubham-Kumar-2000) [reply src/utils/replaceWithExactPath.js:5]: Also there can be new line i.e. `\n` 
Please take these also into account

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]

File: src/utils/replaceWithExactPath.js (hunk 1 above)
Comment (Grayseon) [reply src/utils/replaceWithExactPath.js]: Got it, fixed

File: test/exactPathTest.js [hunk 2]
```diff
@@ -5,7 +5,8 @@ const ProtocolRegistry = require('../src');
 console.log('Registering...');
 ProtocolRegistry.register({
     protocol: 'testproto',
-    command: `node ${path.join(__dirname, './speak.js')} $_URL_`,
+    command: `node ${path.join(__dirname, './speak.js')} $_URL_
+    say running; say running2`,
```
Comment (Shubham-Kumar-2000) [inline test/exactPathTest.js:9]: remove debugging code

File: src/utils/replaceWithExactPath.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const which = require('which');
+
+function replace(command) {
+    return command
+        .split(/&&|;|\n/)
```
Comment (Shubham-Kumar-2000) [inline src/utils/replaceWithExactPath.js:5]: `|` or `||` operators can also as  command separators
Also any of these operators used inside quotes should be ignored
also new lines after `\`  operator should be ignored

**Review by Shubham-Kumar-2000 — CHANGES_REQUESTED** [review]

File: src/utils/replaceWithExactPath.js (hunk 3 above)
Comment (Shubham-Kumar-2000) [reply src/utils/replaceWithExactPath.js:5]: Please explore these further for any more such exceptions

Comment (Grayseon) [reply src/utils/replaceWithExactPath.js]: Fixed

Comment (Grayseon) [reply src/utils/replaceWithExactPath.js]: Hi, please review the changes I have made. I had no idea shell had so many separators haha! Thanks!

**Comment by Grayseon** [comment]
Hi, sorry to bother you @Shubham-Kumar-2000 but could you please review this PR? Thanks!

**Comment by Shubham-Kumar-2000** [comment]
Hi, @Grayseon 

Sorry I was going through a lot of things this year 
Thus, I couldn't post a reply here

Anyway I checked your PR I do think a simple ` /(?<!\\)(?:[&|;]|&&|\|\||\n)(?=(?:(?:[^"\\]|\\.)|"(?:[^"\\]|\\.)*")*$)/` regex will be a full proof solution here.

We should look into something like [Bash Parser](https://www.npmjs.com/package/bash-parser) and then use a visitor to update the script.

**Comment by Grayseon** [comment]
Hi, @Shubham-Kumar-2000 Thanks for reviewing my PR. No problem, hope everything is going well with you! We can totally use a 3rd-party module if you would like to go that route. The only requirement is that we would be able to read the command used, change it to the absolute path, and re-structure the commands and arguments all back into one. 😄

**Comment by Shubham-Kumar-2000** [comment]
Ok, let me work on this this week.
Also for this I want to add bunch of test cases

**Comment by Shubham-Kumar-2000** [comment]
Fixed in v1.6.0

**Pull request closed without merging**